bucoffea/plot/studies/lo_vs_nlo/lo_vs_nlo.py

```diff
@@ -101,9 +111,18 @@ def plot(inpath):
 
                 # Loop over the distributions
                 for distribution in plotset.keys():
-                    if not distribution in acc.keys():
+                    # Load from cache
+
+                    if not (distribution in merged):
+                        acc.load(distribution)
+                        acc[distribution] = merge_extensions(acc[distribution], acc, reweight_pu=not ('nopu' in distribution))
+                        scale_xs_lumi(acc[distribution])
+                        acc[distribution] = merge_datasets(acc[distribution])
+                        merged.add(distribution)
+                    if not (distribution in acc.keys()):
```

Comment:
117: why are acc.load called twice? both in this loop and at lines 95-98
122: shouldn't the check of acc.keys be before accessing the key like at 118?

Reply:
This script does two things:

1. it makes ratio plots, which are always based on recoil.

2. it makes plots of all kinds of different distributions.

The first load retrieves the recoil distribution for the ratio plots. Merging, scaling, rebinning, etc happen inside the ratio plot function. Since we only do this a few times, we don't need to optimize further.

The second load then loads whatever distribution is currently to be plotted (could be recoil, could be something else).

Reply:
Regarding the key check you are right, it's in the wrong place. Fixed now.